### PR TITLE
ASIT-61: Added a feature to send/delete scheduled message

### PR DESCRIPTION
**Pull request by dmitrii-kovtunov**

### CODE REVIEW

File: src/com/messagemedia/clientgui/Manager.java [hunk 1]
```diff
@@ -116,6 +120,7 @@ public static String sendMessage(SendPanel panel){
     	String to        = panel.getToField().getText();
     	String origin    = panel.getOriginField().getText();
     	boolean delivery = panel.getDeliveryReportBox().isSelected();
+		XMLGregorianCalendar scheduled = null;
```
Comment (elruwen) [inline src/com/messagemedia/clientgui/Manager.java]: Can you adjust it to the existing layout?

File: src/com/messagemedia/clientgui/Manager.java [hunk 2]
```diff
@@ -131,7 +136,16 @@ public static String sendMessage(SendPanel panel){
 				validity = Short.parseShort(panel.getValidityPeriodField().getText());
 			}catch(NumberFormatException e){}
 		}
-		
+
+		if (!panel.getScheduledField().getText().equals("")) {
```
Comment (elruwen) [inline src/com/messagemedia/clientgui/Manager.java]: Can you turn the comparision around? If text is null, you will get a NPE

File: src/com/messagemedia/clientgui/Manager.java [hunk 3]
```diff
@@ -131,7 +136,16 @@ public static String sendMessage(SendPanel panel){
...
+			try {
+				Date date = DATE_FORMAT.parse(panel.getScheduledField().getText());
+				GregorianCalendar gregorianCalendar = (GregorianCalendar) GregorianCalendar.getInstance();
+				gregorianCalendar.setTime(date);
+				scheduled = DatatypeFactory.newInstance().newXMLGregorianCalendar(gregorianCalendar);
+			} catch (Exception e) {}
```
Comment (elruwen) [inline src/com/messagemedia/clientgui/Manager.java]: And print at least a stacktrace to stderr?

File: src/com/messagemedia/clientgui/Manager.java [hunk 4]
```diff
@@ -164,8 +183,27 @@ public static String sendMessage(SendPanel panel){
...
     	}
-    } 
-    
+    }
+
+	public static String deleteScheduledMessage(SendPanel panel) {
+		try {
+			long uid = 0;
+
+			if (!panel.getUidField().getText().equals("")) {
```
Comment (elruwen) [inline src/com/messagemedia/clientgui/Manager.java]: See above

File: src/com/messagemedia/clientgui/Manager.java (hunk 3 above)
Comment (dmitrii-kovtunov) [inline src/com/messagemedia/clientgui/Manager.java]: Why should I do this if other pieces of code do not do this? I wanted consistency at least.

File: src/com/messagemedia/clientgui/Manager.java (hunk 4 above)
Comment (dmitrii-kovtunov) [inline src/com/messagemedia/clientgui/Manager.java]: I'll reformat the file using just spaces because now there is a mixture of tabs and spaces.

Comment (elruwen) [inline src/com/messagemedia/clientgui/Manager.java]: I am talking about the way you compare? If text is null you get a NPE.

Comment (dmitrii-kovtunov) [inline src/com/messagemedia/clientgui/Manager.java]: Ah, i thought you mean indentation.

Comment (dmitrii-kovtunov) [inline src/com/messagemedia/clientgui/Manager.java]: done

**Pull request merged**